delegated-events.js

```diff
@@ -14,15 +15,20 @@ function before(subject, verb, fn) {
   return subject;
 }
 
-function matches(selectors, target) {
+function matches(selectors, target, reverse) {
   const queue = [];
   let node = target;
 
   do {
     if (node.nodeType !== 1) break;
     const matches = selectors.matches(node);
     if (matches.length) {
-      queue.push({node: node, observers: matches});
+      const m = {node: node, observers: matches};
+      if (reverse) {
+        queue.unshift(m);
+      } else {
+        queue.push(m);
```

Comment:
For the capture phase, we walk down the tree rather than up. So the resulting match set should be in reverse order.